src/helpers/handlers/index.ts

```diff
@@ -12,22 +13,20 @@ declare global {
   }
   const mapboxgl: typeof mapboxgl;
 }
-export interface Feature {
+/** FeatureWithLayer extends Feature
+ * Adds layer and geometry as Point
+ * Geometry in Feature has GeometryCollection which has no coordinates prop hence  we use geometry as Point
+ */
+export interface FeatureWithLayer extends Feature {
   layer: FlexObject;
-  properties: FlexObject;
-  source: string;
-  state: FlexObject;
-  type: string;
-  geometry?: FlexObject;
-  vectorTileFeature?: FlexObject;
+  geometry: Point;
```

Comment:
Doesn't the default implementation of `Feature` include geometry, which accepts points among other things?  What is the reason for changing it so that it is limited to just `Point`?

Reply:
True but It also accepts GeometryCollection which has no coordinates. I've included some docstrings to inform on this. I am overriding the geometry prop and strictly using Point since that is what we are dealing with in this scenario. What would be the best approach to this scenario ?

Reply:
Could you explain the problem in a little more detail?